### PR TITLE
[0.7.0]: Validate setup data

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Adds validation for the setup data, so that we fail early if necessary data are not provided instead of waiting for the first round to happen, and we give an informative error.

Moreover, removes the FFW round because it is not meaningful now that the setup data cannot be empty.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/models.py [hunk 1]
```diff
@@ -126,13 +127,26 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.setup_params = setup_params
         super().__init__(*args, **kwargs)
 
+        if not self.context.is_abstract_component:
+            # setup data are mandatory for non-abstract skills,
+            # and they should always contain at least the `safe_contract_address`
```
Comment (DavidMinarsch) [inline packages/valory/skills/abstract_round_abci/models.py:132]: And `all_participants`. Please discuss with Viraj how we can configure CLI to directly pass both values in. For e2e tests we will have to have some changes to in order to properly populate this.

File: packages/valory/skills/registration_abci/rounds.py [hunk 2]
```diff
@@ -64,23 +59,16 @@ class RegistrationStartupRound(CollectSameUntilAllRound):
...
-                synchronized_data_class=self.synchronized_data_class,
-            )
-            return synchronized_data, Event.DONE
-        return None
+        if not self.collection_threshold_reached:
+            return None
+
+        synchronized_data = self.synchronized_data.update(
+            participants=frozenset(self.collection),
+            all_participants=frozenset(self.collection),
```
Comment (DavidMinarsch) [inline packages/valory/skills/registration_abci/rounds.py:67]: This needs to change. As we know `all_participants` from the start now.

File: packages/valory/skills/abstract_round_abci/models.py (hunk 1 above)
Comment (Adamantios) [reply packages/valory/skills/abstract_round_abci/models.py:132]: We would simply need to add `all_participants` to the `setup` params in the config and create a command like `_try_update_safe_contract_address` for `all_participants`:

https://github.com/valory-xyz/open-autonomy/blob/6395ee932295b185597f64a3934eeae56c0ae978/autonomy/deploy/base.py#L229

**Comment by Adamantios** [comment]
e2e failure does not make much sense:

```
Error: Package loading error: An error occurred while loading skill valory/registration_abci:0.1.0: Traceback (most recent call last):
  File "/tmp/tmpur1u9xt_/agent_00003/vendor/valory/skills/abstract_round_abci/models.py", line 144, in _ensure_setup
    enforce(bool(self.setup_params), "`setup` params contain no values!")
  File "/home/runner/work/open-autonomy/open-autonomy/.tox/packages-py3.10-linux/lib/python3.10/site-packages/aea/exceptions.py", line 106, in enforce
    raise exception_class(exception_text)
aea.exceptions.AEAEnforceError: `setup` params contain no values!
```

However the `registration_abci` skill is abstract:

https://github.com/valory-xyz/open-autonomy/blob/8be71a59f41c7bdd896a6338ab059b7a169ba1ce/packages/valory/skills/registration_abci/skill.yaml#L130

and therefore should not execute a check for the `setup` params:

https://github.com/valory-xyz/open-autonomy/blob/5c37dc902fd7a5a896c43b464b88e95b921dc16c/packages/valory/skills/abstract_round_abci/models.py#L130-L133

File: packages/valory/skills/abstract_round_abci/models.py (hunk 1 above)
Comment (DavidMinarsch) [reply packages/valory/skills/abstract_round_abci/models.py:132]: I suggest you do this in a new PR on top - it will affect a lot of tests

Comment (Adamantios) [reply packages/valory/skills/abstract_round_abci/models.py:132]: Done in https://github.com/valory-xyz/open-autonomy/pull/1677.

**Review by DavidMinarsch — APPROVED** [review]

**Review by 0xArdi — APPROVED** [review]

**Pull request merged**